File: backend/backoffice-old/src/main/java/com/github/seungyeop_lee/book_rental_shop/backoffice/member/domain/Member.java
package com.github.seungyeop_lee.book_rental_shop.backoffice.member.domain;

import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "MEMBER")
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MEMBER_ID")
    protected Long id;

    protected String name;

    @Column(unique = true)
    protected String phoneNumber;
}

File: backend/backoffice/context/member/domain/member.go
package domain

import "github.com/seungyeop-lee/book-rental-shop/backend/backoffice/context/member/vo"

type Member struct {
	Id          vo.MemberId
	Name        string
	PhoneNumber string
}

func NewMember(name string, phoneNumber string) *Member {
	return &Member{
		Id:          vo.EmptyMemberId,
		Name:        name,
		PhoneNumber: phoneNumber,
	}
}
